src/main/java/org/shanerx/tradeshop/shop/Shop.java

```diff
@@ -0,0 +1,1211 @@
+	 * Fixes values and objects after loading or creating a Shop
+	 */
+	public void fixAfterLoad() {
+		if (utils == null)
+			utils = new Utils();
+
+		shopLoc.stringToWorld();
+		if (!getShopType().isITrade() && chestLoc != null) {
+			chestLoc.stringToWorld();
+			cost.removeIf(item -> item.getItemStack().getType().toString().endsWith("SHULKER_BOX") && getInventoryLocation().getBlock().getType().toString().endsWith("SHULKER_BOX"));
+			utils.PLUGIN.getDataStorage().addChestLinkage(chestLoc, shopLoc);
+		}
+
+		setShopSettings();
+
+		fixSide(ShopItemSide.COST);
+		fixSide(ShopItemSide.PRODUCT);
+
+		if (getShopSign() != null)
+			updateSign();
+	}
+
+	/**
+	 * Generates a semi readable output of the shop object for debug output
+	 *
+	 * @return Return String for debug output
+	 */
+	public String toDebug() {
+		StringBuilder sb = new StringBuilder();
+		sb.append("Shop Debug: \n");
+		sb.append("Shop Chunk: ").append(new ShopChunk(shopLoc.getChunk()).serialize()).append("\n");
+		sb.append("Sign Location: ").append(shopLoc.serialize()).append("\n");
+		sb.append("Shop Type: ").append((isMissingItems() ? Setting.SHOP_INCOMPLETE_COLOUR : Setting.SHOP_GOOD_COLOUR).getString() + shopType.toHeader()).append("\n");
+		sb.append("Shop Status: ").append(status.getLine()).append("\n");
+		sb.append("Storage Location: ").append(hasStorage() ? getInventoryLocationAsSL().serialize() : "N/A").append("\n");
+		sb.append("Storage Type: ").append(hasStorage() ? getStorage().getType().toString() : "N/A").append("\n");
+		sb.append("Owner: ").append(owner.getName()).append(" | ").append(owner.getUUID()).append("\n");
+		sb.append("Managers: ").append(managers.isEmpty() ? "N/A" : managers.size()).append("\n");
+		if (!managers.isEmpty())
+			getUsers(ShopRole.MANAGER).forEach(manager -> sb.append("          ").append(manager.getName()).append(" | ").append(manager.getUUID()).append("\n"));
+		sb.append("Members: ").append(members.isEmpty() ? "N/A" : members.size()).append("\n");
+		if (!members.isEmpty())
+			getUsers(ShopRole.MEMBER).forEach(member -> sb.append("         ").append(member.getName()).append(" | ").append(member.getUUID()).append("\n"));
+		sb.append("Products: ").append(product.isEmpty() ? "N/A" : product.size()).append("\n");
+		if (!product.isEmpty())
+			product.forEach(productItem -> sb.append("          ").append(productItem.toConsoleText()).append("\n"));
+		sb.append("Costs: ").append(cost.isEmpty() ? "N/A" : cost.size()).append("\n");
+		if (!cost.isEmpty())
+			cost.forEach(costItem -> sb.append("       ").append(costItem.toConsoleText()).append("\n"));
+
+		return utils.colorize(sb.toString());
+	}
+
+	/**
+	 * Saves the shop to file
+	 */
+	public void saveShop() {
+		updateFullTradeCount();
+		utils.PLUGIN.getDataStorage().saveShop(this);
+		updateUserFiles();
+	}
+
+	/**
+	 * Saves the shop to file if passed boolean is true
+	 *
+	 * @param shouldSave true if save should proceed
+	 */
+	public void saveShop(boolean shouldSave) {
```

Comment:
this should be removed; just use `if (shouldSave) saveShop();`

Reply:
Looks like I was thinking of the config save rather than this one. 

I do still prefer this where it is used and will leave it in.